src/Api/Persistent.hs

```diff
@@ -159,6 +160,14 @@ getSpaces = getDb dbSpaces
 getIdeas :: Persist [Idea]
 getIdeas = getDb dbIdeas
 
+-- | If idea space already exists, return 'False'.  Otherwise, create it and return 'True'.
+addIdeaSpace :: IdeaSpace -> Persist Bool
```

Comment:
yes, but i don't want exceptions here.  even if we had them i didn't want them.

are you sure you do?  if so perhaps we should talk about this tomorrow.